jax/_src/pjit.py

```diff
@@ -1442,7 +1446,8 @@ def _pjit_lower_cached(
         jaxpr.in_avals, in_is_global=in_is_global, keep_unused=keep_unused,
         always_lower=always_lower,
         devices_from_context=(
-            None if mesh is None or mesh.empty else list(mesh.devices.flat)))
+            None if mesh is None or mesh.empty else list(mesh.devices.flat)),
+        lowering_platform=lowering_platform)
```

Comment:
I don't know how your tests passes but you read this in the wrong order in pxla.py in lower_sharding_computation

Reply:
This works because there is a "*" and it forces those arguments to work as kwargs and then the order does not matter.

Reply:
Ooff.. that's confusing. Can we keep the `*` change to a separate CL and not do it here?

Reply:
I change the order at the function definition for clarity.